services/wireguard/resources/allocator.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package resources
+
+import "net"
+
+// Allocator is mock wireguard resource handler.
+// It will manage lists of network interfaces names, IP addresses and port for endpoints.
+type Allocator struct{}
+
+// AllocateInterface provides available name for the wireguard network interface.
+func (h *Allocator) AllocateInterface() string { return "myst0" }
```

Comment:
not to be nitpicky but can we refactor this  (h *Allocator) to an (a *Allocator)? :)

Reply:
we definitely need to do this, thank.

Reply:
Fixed and rebased to latest master.